brain_friendly.py

```diff
@@ -1,13 +1,16 @@
 # The input BF program.
-bf_program_string = "++++++>----"
+bf_program_string = '[][]'
 # Current pointer location.
 bf_index = 0
 # Array of byte cells (currently 30,000).
 bf_array = [0]*30000
 
-
-def eval_program(program_string, index, array, input=None, output=None):
-    for command in program_string:
+def eval_program(program, index, array, input=None, output=None):
+    program_index = 0
+    # Stack for "[" and "]" commands.
+    brace_dict = get_brace_matches(program)
```

Comment:
`brace_dict` isn't the most descriptive name, `brace_associations` might be better? `brace_pairs`?